src/coreComponents/mesh/WellElementRegion.cpp

```diff
@@ -64,7 +64,7 @@ void WellElementRegion::generateWell( MeshLevel & mesh,
 
   globalIndex const matchedPerforations = MpiWrapper::sum( perforationData->size() );
   GEOS_THROW_IF( matchedPerforations != numPerforationsGlobal,
-                 "Invalid mapping perforation-to-element in well " << lineBlock.getName() << "." <<
+                 "Invalid mapping perforation-to-element in InternalWell " << lineBlock.getDataContext() << "." <<
```

Comment:
As a matter of fact `InternalWell` should not be hard coded here.
Do you have a way to get the name of the xml node instead?
Otherwise it's OK.

Reply:
I used in my last refactoring commit `InternalWellGenerator::catalogName()`.
After testing it I noticed there is no useful line to show here. I think it would be better to use getWellGeneratorName() here (nearer of the problematic perforation place).
```suggestion
                 "Invalid mapping perforation-to-element in "<<
                 InternalWellGenerator::catalogName() << " " << getWellGeneratorName() << "." <<
```

As this `wellGeneratorName` property is only used to output messages, could we change it to the `DataContext::toString()` value, or to a pointer to a `DataContext`? This way we would also have the `InternalWell` line number.